### PR TITLE
minivmac: init at 2024.06.08

### PR DESCRIPTION
**Pull request by flokli**
## Things done

- Built on platform(s)
  - [x] x86_64-linux (and cross-compiled to aarch64-linux)
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mi/minivmac/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  buildInputs = [ xorg.libX11 ];
+
+  makeFlags = [ "PREFIX=${placeholder "out"}" ];
+
+  configurePhase = ''
+    ${buildPackages.stdenv.cc}/bin/cc setup/tool.c -o setup_t
+    ./setup_t -t lx64 > setup.sh
+
+    # Patch hardcoded references in setup.sh to cross-aware counterparts
+    sed -i -e "s|gcc |${stdenv.cc.targetPrefix}cc |" setup.sh
```
Comment (SomeoneSerge) [inline pkgs/by-name/mi/minivmac/package.nix:30]: nits: if you're not relying on regexps you might as well use substituteInPlace instead

File: pkgs/by-name/mi/minivmac/package.nix
Comment (SomeoneSerge) [inline pkgs/by-name/mi/minivmac/package.nix:1]: More nits per usual: explicit `getExe'`, `getLib`, etc are in my opinion preferable.

Comment (SomeoneSerge) [reply pkgs/by-name/mi/minivmac/package.nix:1]: @ofborg eval

The error seems unrelated https://gist.github.com/GrahamcOfBorg/a2f3293c3e5c1afe52894b05a5909294

**Comment by emilazy** [comment]
Is there a canonical upstream here? I know that the original maintainer Paul C. Pratt has been inactive of late, but it seems like the GitHub repository this one is forked from has features that this WASM‐oriented fork doesn’t, and vice versa. Is this fork widely used outside of Infinite Mac?

(I think it’s totally fine to package this regardless, but I’m just wondering if we should maybe append something to the name to make it clear that the provenance might not match what your expectation for “Mini vMac” is.)

**Comment by flokli** [comment]
> Is there a canonical upstream here? I know that the original maintainer Paul C. Pratt has been inactive of late, but it seems like the GitHub repository this one is forked from has features that this WASM‐oriented fork doesn’t, and vice versa. Is this fork widely used outside of Infinite Mac?
> 
> (I think it’s totally fine to package this regardless, but I’m just wondering if we should maybe append something to the name to make it clear that the provenance might not match what your expectation for “Mini vMac” is.)

The "what is upstream" question seems to be a bit more complicated. I'd be happy to use a suffix or prefix in pname or version to point it out more what we're using here. Any suggestions?

File: pkgs/by-name/mi/minivmac/package.nix
Comment (flokli) [reply pkgs/by-name/mi/minivmac/package.nix:1]: Done.

File: pkgs/by-name/mi/minivmac/package.nix (hunk 1 above)
Comment (flokli) [reply pkgs/by-name/mi/minivmac/package.nix:30]: Done

**Comment by flokli** [comment]
On the fork situation: I added a "-erichelgeson" to the `pname`, and updated the desription (but decided to keep it at the `minivmac` attribute name.

**Pull request merged**